sklearn/utils/tests/test_weight_vector.py

```diff
@@ -0,0 +1,23 @@
+# Author: Mathis Batoul <batoulmathis@gmail.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from sklearn.utils._weight_vector import (
+    WeightVector32,
+    WeightVector64,
+)
+
+
+def test_type_invariance():
```

Comment:
You should be able to parametrise this test on the `dtype` using [`pytest.mark.parametrise`](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) to remove duplication.

Reply:
A comment to describe the goal of the test is also welcomed